src/daemon/scheduler.rs

```diff
@@ -169,230 +126,161 @@ fn make_cas_event_triggers(event_queue: Arc<MessageQueue>, ca_manager: Arc<CaMan
+        // to avoid a thundering herd. Note that the operator can always
+        // choose to run bulk operations manually if they know that they
+        // cannot wait.
+        let ca_list = self.ca_manager.ca_list(&self.system_actor)?;
+        let cas = ca_list.cas();
+
+        debug!("Adding tasks at start up");
+
+        let mut use_jitter = cas.len() >= SCHEDULER_USE_JITTER_CAS_THRESHOLD;
+
+        for summary in cas {
+            let ca = self.ca_manager.get_ca(summary.handle()).await?;
+
+            use_jitter = use_jitter || ca.nr_parents() >= self.config.ca_refresh_parents_batch_size;
+
+            debug!("Adding tasks for CA {}, using jitter: {}", ca.handle(), use_jitter);
+
+            // Plan a regular sync for each parent. Spread these out if there
+            // are too many CAs or parents for a CA. In cases where there are only
+            // a handful of CAs/parents, this 'ca_refresh_start_up' will be 'now'.
+            // Note: users can override using the 'bulk' functions.
+            for parent in ca.parents() {
+                self.tasks.sync_parent(
+                    ca.handle().clone(),
+                    parent.clone(),
+                    self.config.ca_refresh_start_up(use_jitter),
+                );
             }
-        })
-    })
-}
-
-fn make_cas_refresh(ca_server: Arc<CaManager>, refresh_rate: u32, actor: Actor) -> ScheduleHandle {
-    let server_started = Timestamp::now();
-    SkippingScheduler::run(refresh_rate, "CA certificate refresh", move || {
-        let rt = Runtime::new().unwrap();
-        rt.block_on(async {
-            debug!("Triggering background refresh for all CAs");
-            ca_server.cas_refresh_all(server_started, &actor).await;
-        });
-    })
-}
 
-fn make_announcements_refresh(bgp_analyser: Arc<BgpAnalyser>) -> ScheduleHandle {
-    SkippingScheduler::run(5, "update RIS BGP info", move || {
-        let rt = Runtime::new().unwrap();
-        rt.block_on(async {
-            if let Err(e) = bgp_analyser.update().await {
-                error!("Failed to update BGP announcements: {}", e)
+            // Plan a sync with the repo. In case we only have a handful of CAs
+            // then the result is that the sync is scheduled asap. Otherwise
+            // spread the load.
+            // Note: if circumstances dictate a sync before it's planned, e.g.
+            // because ROAs are changed, then it will be rescheduled accordingly.
+            // Note: users can override using the 'bulk' functions.
+            self.tasks
+                .sync_repo(ca.handle().clone(), self.config.ca_refresh_start_up(use_jitter));
+
+            // If suspension is enabled then plan a task for it. Since this is
+            // a cheap no-op in most cases, we do not need jitter. If we do not
+            // add this task then it will not be executed (obviously), but more
+            // importantly.. by adding this task we ensure that it will keep being
```

Comment:
I can't find in the previous code where child suspension was done periodically, only on CA refresh. What's the rationale for adding it as a recurring background task now?

Reply:
Previously 'ca refresh' dit two things:
- refresh this CA with its parent
- suspend children (if enabled)

Now these are separate tasks